src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -2087,6 +2095,57 @@ private BoundExpression MakeNullCheck(SyntaxNode syntax, BoundExpression rewritt
                     type);
         }
 
+        private BoundExpression RewriteBuiltInNativeShiftOperation(
+            BoundBinaryOperator? oldNode,
+            BinaryOperatorKind operatorKind,
+            BoundExpression loweredLeft,
+            BoundExpression loweredRight,
+            TypeSymbol type)
+        {
+            // For the predefined operators, the number of bits to shift in `x << count`, `x >> count` or `x >>> count` is computed as follows:
+            // [...]
+            // - When the type of `x` is `nint` or `nuint`, the shift count is given by
+            //   the low-order five bits of `count` on a 32 bits platform, or
+            //   the lower-order six bits of `count` on a 64 bits platform.
+            //   The shift count is computed from `count & (sizeof(nint) * 8 - 1)` (or `nuint`),
+            //   which is `count & 0x1F` on a 32 bits platform and `count & 0x3F` on a 64 bits platform.
+
+            Debug.Assert(loweredLeft.Type is { });
+            TypeSymbol leftType = loweredLeft.Type;
+            Debug.Assert(leftType.SpecialType is SpecialType.System_IntPtr or SpecialType.System_UIntPtr);
+
+            ConstantValue? rightConstantValue = loweredRight.ConstantValue;
+            Debug.Assert(loweredRight.Type is { });
+            TypeSymbol rightType = loweredRight.Type;
+            Debug.Assert(rightType.SpecialType == SpecialType.System_Int32);
+
+            if (rightConstantValue != null && rightConstantValue.IsIntegral && rightConstantValue.Int32Value <= 0x1F)
```

Comment:
>rightConstantValue.Int32Value <= 0x1F [](http://example.com/codeflow?start=79&length=37)

Are we checking `rightConstantValue >= 0`?

Reply:
Thanks. Added